.github/workflows/test.yml

```diff
@@ -7,22 +7,140 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        flask-version: ['1.0', '1.1', '2.0', '2.1', 'latest']
         os: [ubuntu-18.04, macos-latest, windows-latest]
-        flask-version: ["1.0", 1.1, "2.0", latest]
+        exclude:
+          # starting from Flask 2.1.0, python 3.6 is no longer supported:
+          - flask-version: '2.1'
+            python-version: '3.6'
+
+          - flask-version: 'latest'
+            python-version: '3.6'
+
+          # old versions of Flask no longer working with python 3.10:
+          - flask-version: '1.0'
+            python-version: '3.10'
+
+          # old versions of Flask don't install without some hacks in shell:
+          - flask-version: '1.0'
+            os: windows-latest
+
+          - flask-version: '1.1'
+            os: windows-latest
+
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+
       - name: Install poetry
-        run: pip install poetry
-      - name: Force the chosen flask version
-        run: poetry add flask==${{ matrix.flask-version }}
+        run: pip3 install poetry
+
+      - name: Force Python version (linux/macOS)
+        if: matrix.os != 'windows-latest'
+        # sed command for macOS: https://stackoverflow.com/a/44864004
+        run: sed -i.bak 's/python = "^3.6"/python = "~${{ matrix.python-version }}"/' pyproject.toml
+
+      - name: Force Python version (windows)
+        if: matrix.os == 'windows-latest'
+        run: (Get-Content pyproject.toml).replace('python = "^3.6"', 'python = "~${{ matrix.python-version }}"') | Set-Content pyproject.toml
+
+      - name: Force Flask version
+        run: poetry add Flask==${{ matrix.flask-version }}
+
+      - name: Overwrite Flask dependencies for legacy install
+        if: matrix.flask-version < '2.0' && matrix.flask-version != 'latest'
```

Comment:
some packages that are now too recent when installed via the Flask 1.X dependencies